machinator-scala/src/Machinator/Scala/Scheme/Circe/Codegen.hs

```diff
@@ -131,6 +140,39 @@ generateFromJsonV1 (M.Definition (M.Name tn) _ typ) =
             )
       )
 
+
+-- | Generate a Circe KeyEncoder (for newtyped ground types only).
+generateToJsonKeyV1 :: M.Definition -> Doc a
+generateToJsonKeyV1 def@(M.Definition (M.Name tn) _ (M.Newtype (M.Name wrapper, wrappedType@(M.GroundT _)))) =
+  text "implicit val"
+    <+> text tn <> text "KeyEncoder" <> text ":"
+    <+> text "io.circe.KeyEncoder" <> WL.brackets (typeNameV1 def)
+    <+> text "="
+    <+> code_block
+      [ case_expr
+          (text tn <> WL.tupled [text wrapper])
+          (text "io.circe.KeyEncoder" <> WL.brackets (genTypeV1 wrappedType) <> text ".apply" <> WL.parens (text wrapper))
+      ]
+generateToJsonKeyV1 _ = WL.mempty
+
+-- | Generate a Circe KeyDecoder (for newtyped ground types only).
+generateFromJsonKeyV1 :: M.Definition -> Doc a
+generateFromJsonKeyV1 def@(M.Definition (M.Name tn) _ (M.Newtype (_, M.GroundT ty))) =
+    fromMaybe WL.mempty $ with (parse ty) $ \parser ->
```

Comment:
To make it clearer what's going on here, I would use a pattern guard
```haskell
  | Just parser <- parse ty
  = ...
```
As currently there are two separate ways it can be a mempty.